### PR TITLE
Update with rustfmt, clippy and add to CI

### PR DESCRIPTION
**Pull request by pjsier**
Should finish #12. Makes updates recommended by clippy and rustfmt and also adds them to GitHub Actions checks

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -59,3 +59,47 @@ jobs:
...
+      - name: Install clippy
+        run: rustup component add clippy
+
+      - name: Run cargo clippy
+        uses: actions-rs/cargo@v1
+        with:
+          command: clippy
+          args: --all-targets --all-features -- -D warnings
+
+  fmt:
```
Comment (michaelkirk) [inline .github/workflows/test.yml:86]: I'm going to steal this and try to foist it on every project I can. 👏 👏 👏

**Comment by urschrei** [comment]
Does this fail a test on clippy / rustfmt failures, or just warn with details?

**Comment by pjsier** [comment]
Right now it fails, but I think it would make sense for either of those checks to be a warning instead

**Comment by urschrei** [comment]
Ah. I ask because there's been some tension regarding formatting requirements (but not Clippy, actually) being seen as a cause of friction / contributor barrier to entry on the actual Rust project, so I was curious. One thing that was pointed out was that the formatting run can be automated to occur before a merge commit, which I like a lot, but have no idea how to do. I like the idea of a warning for either / both tools, though, and we should discuss adopting that more widely if people are amenable.

**Comment by pjsier** [comment]
That makes sense! I usually have my editor set up to run rustfmt on save, but I can imagine for clippy in particular that might be more difficult.

For running before a merge commit, I think with GitHub Actions now you can have a separate user make changes and then commit them to a branch. I'd be a little worried about automated commits for the formatter, but that isn't really based on much and if it becomes an issue I think we could set that up.

Changing these to warn instead of failing seems like a good next step for now, and I can update that here, thanks for the comments!

**Comment by pjsier** [comment]
Any other thoughts on this before I merge?

**Review by frewsxcv — APPROVED** [review]

**Pull request merged**